stdlib/test/builtin/test_any_all.mojo

```diff
@@ -0,0 +1,67 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+# RUN: %mojo %s
+
+from testing import assert_true, assert_false
+from collections.set import Set
+
+
+def test_list_any():
+    var all_true = List(1, 2, 3, 4)
```

Comment:
nit: Can you please add some tests with different types here and below (e.g. bool and some float type)

Reply:
good idea

Reply:
i also need to test empty iterables